src/Domain/CoordinateMaps/FocallyLiftedMapHelpers.hpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/TypeTraits/RemoveReferenceWrapper.hpp"
+
+/// Holds helper functions for use with
+/// domain::CoordinateMaps::FocallyLiftedMap.
+namespace domain::CoordinateMaps::FocallyLiftedMapHelpers {
+
+/*!
+ * \brief Finds how long to extend a line segment to have it intersect
+ * a point on a 2-sphere.
+ *
+ * \details Consider a 2-sphere with center \f$C\f$ and radius \f$R\f$, and
+ * and let \f$P\f$ and \f$x_0\f$ be two arbitrary 3D points.
+ *
+ * Consider the line passing through \f$P\f$ and \f$x_0\f$.
+ * If this line intersects the sphere at a point \f$x\f$, then we can write
+ *
+ * \f[
+ *  x = P + (x_0-P) \lambda,
+ * \f]
+ *
+ * where \f$\lambda\f$ is a scale factor.
+ *
+ * `scale_factor` computes and returns \f$\lambda\f$.
+ *
+ * ### Even more detail:
+ *
+ * To solve for \f$\lambda\f$, we note that \f$x\f$ is on the surface of
+ * the sphere, so
+ *
+ * \f[
+ *  |x-C|^2 = R^2,
+ * \f]
+ *
+ *  or equivalently
+ *
+ * \f[
+ *  | P-C + (x_0-P)\lambda |^2 = R^2.
+ * \f]
+ *
+ * This is a quadratic equation for \f$\lambda\f$
+ * and it generally has more than one real root.
+ *
+ * So how do we choose between multiple roots?  Some of the maps that
+ * use `scale_factor` assume that *for all points*, \f$x_0\f$ is
+ * between \f$P\f$ and \f$x\f$.  Those maps should set the parameter
+ * `src_is_between_proj_and_target` to true. Other maps assume that
+ * *for all points*, \f$x\f$ is always between src_point
+ * and proj_center. Those maps should set the parameter
+ * `src_is_between_proj_and_target` to false.
+ *
```

Comment:
Please clarify here that proj_center is P, sphere_center is C, src_point is x_0 (is it?), and intersection_point is x_1 (is it), so you can go between the equations and the inputs to the functions